### DISCUSSION TITLE
SQL Notebook as .sqlnb file type for store cell results

### DISCUSSION THREAD

**Discussion by miguel-vivanco** [cmoog/vscode-sql-notebook#32] [Ideas]
I guess it would be great to work with a new file type like python notebook (pynb) but for sql, it would be .sqlnb where would stored the cell results and execution time, and maybe in future releases plotted / graphic results. Thanks

**Comment by cmoog** [comment]
Great idea. This would address many of the underlying issues seen in #23 and #10.

**Reply by RollsChris** [reply]
Hey I was thinking the same.. only annoying thing I can think of is registering the the file type with vscode.. icons etc. 

How do we push this forward anyway? I'm up for helping as I'm finding the tool very useful.

**Comment by RollsChris** [comment]
Got a working example using a .sqlnb file type, Can use along side old serializer.. we could also enable as an experimental feature?

Check it out if you like ( it persists output which is the part i really want at the moment): 
https://github.com/CR-OpenSource/vscode-sql-notebook

**Reply by cmoog** [reply]
Oh, nice work. Yeah, I agree that we'd want to release support for `.sqlnb` alongside the existing support for `.sql`. Eventually, we may be able to add an action that converts between the two (of course losing the query output when going from `.sql` -> `.sqlnb`). Seeing as this would be a new filetype, it'd probably be good to document a specification prior to releasing the feature, with design comparisons to python notebooks and other comparable formats informing the spec decisions.

**Reply by RollsChris** [reply]
OK by specification do you mean outline the structure of the json objects: 
```typescript 
interface RawNotebookCell {
  language: string;
  value: string;
  kind: vscode.NotebookCellKind;
  editable?: boolean;
  outputs: RawCellOutput[];
}

interface RawCellOutput {
  mime: string;
  value: any;
}
```
The various uses for each property? 

Where are you placing documentation... there is no wiki at the moment?

**Comment by RollsChris** [comment]
Hey me again lol I've added support for the data-table extension.. and I think it works nicely, it can handle thousands of rows etc allows sorting of columns and more..

 Once we get some momentum ill clean things up but for now I'm just proof of concepting.

![image](https://user-images.githubusercontent.com/6059035/160921542-c4a6a9d0-b7ad-4eab-b8e9-1ea463ba1213.png)


https://github.com/CR-OpenSource/vscode-sql-notebook

**Reply by cmoog** [reply]
Oh very cool, great work @RedEyedDog. I'm certainly open to merging a contribution with that addition.

**Reply by RollsChris** [reply]
Ill clean things up a little and issue a PR then.. id also like to get the sqlnb work in ;)

**Reply by cmoog** [reply]
Please keep the sqlnb changes in a separate branch from the Data Table improvements. I don't think it makes sense to block one on the other.

**Reply by RollsChris** [reply]
OK ill do that

**Comment by RollsChris** [comment]
@cmoog i think this is the next best thing ;)
